api/buildings/static/js/admin/here_map.js

```diff
@@ -0,0 +1,167 @@
+(function($) {
+  // We'll insert the map after this element:
+  var prev_el_selector = "#id_lng";
+
+  // The input elements we'll put lat/lon into and use
+  // to set the map's initial lat/lon.
+  var lat_input_selector = "#id_lat";
+  var lon_input_selector = "#id_lng";
+
+  // The input elements we'll put the address string and country code into.
+  var street_input_selector = "#id_address";
+  var street_number_input_selector = "#id_street_number"
+  var locality_input_selector = "#id_locality";
+
+  // We'll insert the map after this element:
+  var prev_el_selector = "#id_lng";
+
+  // Initial zoom level for the map.
+  var initial_zoom = 12;
+
+  // Initial zoom level if input fields have a location.
+  var initial_with_loc_zoom = 18;
+  var initial_lat = 44.4268;
+  var initial_lon = 26.1025;
+
+  function distance(lat1, lon1, lat2, lon2) {
+    var radlat1 = Math.PI * lat1/180
+    var radlat2 = Math.PI * lat2/180
+    var theta = lon1-lon2
+    var radtheta = Math.PI * theta/180
+    var dist = Math.sin(radlat1) * Math.sin(radlat2) + Math.cos(radlat1) * Math.cos(radlat2) * Math.cos(radtheta);
+    if (dist > 1) {
+        dist = 1;
+    }
+    dist = Math.acos(dist)
+    dist = dist * 180/Math.PI
+    dist = dist * 60 * 1.1515
+    return dist
+  }
+
+  function addDraggableMarker(map, behavior, lat, lng) {
+    var marker = new H.map.Marker({lat:lat, lng:lng}, {
+      // mark the object as volatile for the smooth dragging
+      volatility: true
+    });
+    // Ensure that the marker can receive drag events
+    marker.draggable = true;
+    map.addObject(marker);
+
+    // disable the default draggability of the underlying map
+    // and calculate the offset between mouse and target's position
+    // when starting to drag a marker object:
+    map.addEventListener('dragstart', function(ev) {
+      var target = ev.target;
+      var pointer = ev.currentPointer;
+      if (target instanceof H.map.Marker) {
+        var targetPosition = map.geoToScreen(target.getGeometry());
+        target['offset'] = new H.math.Point(pointer.viewportX - targetPosition.x, pointer.viewportY - targetPosition.y);
+        behavior.disable();
+      }
+    }, false);
+
+    map.addEventListener('dragend', function(ev) {
+      var target = ev.target;
+      if (target instanceof H.map.Marker) {
+        // update the lat and long input fields with the marker's new coordinates
+        var new_lat = Math.round(marker.getGeometry().lat * Math.pow(10, 4)) / Math.pow(10, 4);
```

Comment:
What I presume you attempt here with the multiplication & division is to control the number of digits after the decimal point ?

If that's the case, I believe [`.toFixed()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toFixed) would make it more clear.